pkg/api/validate/dynamic/dynamic.go

```diff
@@ -33,6 +33,13 @@ import (
 	"github.com/Azure/ARO-RP/pkg/util/token"
 )
 
+var (
+	errMsgNSGAttached            = "The provided subnet '%s' is invalid: must not have a network security group attached."
```

Comment:
IMO, it seems odd to create new vars for errors related to BYO NSG, but not for other error messages in this file (from other validators). We should either keep the existing convention, or do this for other validators as well.

Reply:
Done - converted other validators as well.